### ISSUE TITLE
Kotlin-sample

### ISSUE THREAD

**Issue by jmfayard** [wealthfront/magellan#40]
Hello,

I had great success with building an android app with kotlin, leveraging magellan to get rid of activities and fragments, and the additional technique of [abstracting the android widgets with delegated properties](http://marcinmoskala.com/android/kotlin/2017/05/05/still-mvp-or-already-mvvm.html) to make the android views even dumber : [example here](https://github.com/jmfayard/okAndroid/blob/bbdf769a37444352421117c10f000d2c67fa92f5/src/main/kotlin/com/github/jmfayard/okandroid/screens/VerificationScreen.kt#L40-L45) 

Would you like me to contribute an additional sample?

   Jean-Michel

**Comment by fabiendevos** [comment]
Hi @jmfayard ,

You totally can! Ideally, maybe start with something really minimal and simple. I am not really familiar with this technique but this looks really interesting.

Another option which might actually be much better is for you to have a sample in your own repo and I can link to it from the readme or the wiki! That way you can also update it and make it evolve as you want.

Let me know what you prefer,

Fabien

**Comment by jmfayard** [comment]
@fabiendevos 

I would prefer if the sample could live inside the project, but a separate repo is better at first : I'm using android studio canary and beta versions of the android gradle plugin.

I already started, I chose to do a loose reimplementation of the android testing codelab

https://github.com/googlecodelabs/android-testing

My project is here

https://github.com/jmfayard/android-kotlin-magellan

I really like how everything combine. Navigation is a breeze, the android views are Simple and Stupid, mocking just work (kotlin classes are final, but no pb for Mockito with the right configuration!) making unit testing of the screens straightforward.

The two things I want to add next is documentation and instrumentation tests using [Instrumentation Testing Robots](https://news.realm.io/news/kau-jake-wharton-testing-robots/)
